lib/pdn/pdn.c

```diff
@@ -643,8 +643,10 @@ int pdn_dynamic_params_get(uint8_t cid, struct in_addr *dns4_pri,
 		}
 	}
 	if (dns4_sec) {
-		if (zsock_inet_pton(AF_INET, dns4_sec_str, dns4_sec) != 1) {
-			return -EADDRNOTAVAIL;
+		if (matched >= 2) {
+			if (zsock_inet_pton(AF_INET, dns4_sec_str, dns4_sec) != 1) {
+				return -EADDRNOTAVAIL;
+			}
 		}
 	}
 	if (ipv4_mtu) {
```

Comment:
This requires that matched == 3, this will fail to get the mtu with the current changes. We should take into account how many addresses were actually found and get the correct mtu.

Same for IPv6.

Reply:
I didn't understand why will it fail to get mtu. I check >= because it can contain both 2 and 3 case. 2 means only DNS addresses were fetched. 3 means DNS addresses + MTU were fetched. Could you explain me more?

Reply:
Then I think we have two options:
- Seperate these into two function and call CGCONTRDP two times to obtain MTU
- Move MOSH's link_api's link_api_pdp_context_dynamic_params_get into library section to use on both MOSH and SLM, because it was working fine and fulfill our needs
Which one is better?